doc/src/devdocs/compiler.md

```diff
@@ -0,0 +1,119 @@
+# High-level Overview of the Native-Code Generation Process
+
+
+    <placeholder>
```

Comment:
Yeah, I put it there so that nobody could complain that the document was immensely incomplete at describing codegen, and just jumps to explain a few small areas. It almost worked :)